plugin/src/main/java/com/iridium/iridiumskyblock/generators/OceanGenerator.java

```diff
@@ -61,16 +61,78 @@
                 );
 
                 // Generate water or lava on top of the floor
+                XMaterial oceanMaterial = world.getEnvironment() == Environment.NETHER ? XMaterial.LAVA : XMaterial.WATER;
                 for (int y = currentFloorHeight + 1; y <= waterHeight; y++) {
-                    XMaterial material = world.getEnvironment() == Environment.NETHER ? XMaterial.LAVA : XMaterial.WATER;
-                    chunkData.setBlock(x, y, z, Objects.requireNonNull(material.parseMaterial()));
+                    chunkData.setBlock(x, y, z, Objects.requireNonNull(oceanMaterial.parseMaterial()));
                 }
             }
         }
 
         return chunkData;
     }
 
+    public void generateWater(World world, int x, int z) {
+        SimplexOctaveGenerator generator = new SimplexOctaveGenerator(new Random(world.getSeed()), 8);
+        generator.setScale(0.005D);
+
+        XMaterial bottomMaterial = IridiumSkyblock.getInstance().getConfiguration().generatorSettings.oceanFloorBottomMaterial;
+        XMaterial topMaterial = IridiumSkyblock.getInstance().getConfiguration().generatorSettings.oceanFloorTopMaterial;
+        int waterHeight = IridiumSkyblock.getInstance().getConfiguration().generatorSettings.waterHeight;
+        int maxOceanFloorLevel = IridiumSkyblock.getInstance().getConfiguration().generatorSettings.maxOceanFloorLevel;
+        int minOceanFloorLevel = IridiumSkyblock.getInstance().getConfiguration().generatorSettings.minOceanFloorLevel;
+
+        int currentFloorHeight = (int) ((generator.noise(x, z, 1.5D, 0.5D, true) + 1) * (maxOceanFloorLevel - minOceanFloorLevel) + minOceanFloorLevel);
+
+        // Generate layer of bedrock
+        IridiumSkyblock.getInstance().getNms().setBlockFast(
```

Comment:
This wont work properly, We need to delete the blocks that should be air, also for performance we should check if the block were setting isnt already set